geoapps/driver_base/driver.py

```diff
@@ -97,6 +99,14 @@ def start(cls, filepath: str | Path, driver_class=None):
             filepath, validations=driver_class._validations  # pylint: disable=W0212
         )
 
+        version = ifile.data.get("version", None)
```

Comment:
**suggestion:** cannot we guaranty it is backward compatible, and have the warning only if geoapps is older than the version in the file?

otherwise it seems users will get the warning often

Reply:
Two things right now:
1- We have no mechanism to handle backward compatibility set yet.
2- Version is stored as a string with labels like 'alpha', 'rc', so we need to write/find code to interpret that.

Thoughts?

Reply:
Looks like these guys can handle string version comparison: https://packaging.pypa.io/en/latest/index.html

Reply:
* about handling incompatibility: ideally, a new version should not introduce incompatibility with older ui.json. If it does and is not able to patch the ui.json to fix it, it should fail and report clearly about it. Out of scope here. Let's discussion live about it.
* for version comparison, this is semantic versioning: I would suggest using https://python-semver.readthedocs.io/en/latest/

Reply:
Done